client/api/index.js

```diff
@@ -172,12 +173,56 @@ export default class WCStripeAPI {
 			} );
 	}
 
-	confirmIntent( url, savePaymentMethod ) {
+	/**
+	 * Extracts the details about a payment intent from the redirect URL,
+	 * and displays the intent confirmation modal (if needed).
+	 *
+	 * @param {string} redirectUrl The redirect URL, returned from the server.
+	 * @param {string} paymentMethodToSave The ID of a Payment Method if it should be saved (optional).
+	 * @return {string|boolean} A redirect URL on success, or `true` if no confirmation is needed.
+	 */
+	confirmIntent( redirectUrl, paymentMethodToSave ) {
 		console.error( 'TODO: Not implemented yet: confirmIntent' );
 		return true;
 	}
```

Comment:
I copied the full UPE checkout from WC Pay; this function is called from `maybeShowAuthenticationModal()` in `/client/classic/upe/index.js` (which is called on load). We are not using it yet... but I copied the full API interface so it's easier to see what is still left to migrate.

Reply:
I think that when we were separating UPE code from WCPay code, this `maybeShowAuthenticationModal()` function managed to slip through the cracks into the UPE side, because I think none of us were really aware of what it did (I'd never actually seen that authentication modal before, until you mentioned it in your testing instructions!). I just tested this code after commenting out all times that `maybeShowAuthenticationModal()` is called and I _think_ we can safely remove it. It looks like the authentication modal gets created by Stripe when we fire off `confirmPayment`/`confirmSetup` instead and we don't need to create this modal ourselves. 

I'm pretty sure we can remove both this and the `confirmIntent` API function, since they shouldn't be required in our UPE flow. @bborman22, wondering if you perhaps have anymore insight than I do here? Do you know any reason why we need that `maybeShowAuthenticationModal()` function?

Reply:
`maybeShowAuthenticationModal` is only relevant for a card needing authorization. I hadn't realized that UPE was handling that modal now and just ran a couple tests. There was one scenario where the function was still needed, if you save a card that requires authorization, then go to checkout again with that card, this function catches it and requests the authorization. I did this test on WCPay to allow the saved payment flow. There may be a different way to handle this flow, but as it is now, I believe this function still serves that purpose.

Reply:
Ah, gotcha, sounds like I was the only one who had no idea what that function did then. Cool, that makes sense. Let's leave `maybeShowAuthenticationModal` in then for now and we can decide its ultimate fate once we've fleshed out that checkout with saved payments flow.